File: src/main/java/com/projectincremental/entities/Zone.java
package com.projectincremental.entities;

import javax.persistence.*;
import java.util.Set;

@Entity
public class Zone {

	@Id
	@GeneratedValue(strategy=GenerationType.AUTO)
	private Long id;

	private String nom;


	@OneToMany(mappedBy = "zone")
	private Set<Personnage> personnages;

	@OneToMany(mappedBy = "zone")
	private Set<Monstre> monstres;


	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public String getNom() {
		return nom;
	}

	public void setNom(String nom) {
		this.nom = nom;
	}
	
}

File: src/main/ui/src/Models/Zone.ts
import Monstre from "./Monstre";

class Zone {
  id: string;
  nom: string;
  monstres: Monstre[];

  constructor(id: string, nom: string, monstres: Monstre[]) {
    this.id = id;
    this.nom = nom;
    this.monstres = monstres;
  }

  get id(): string {
    return this.id;
  }
  get nom(): string {
    return this.nom;
  }
  get monstres(): string {
    return this.monstres;
  }
}
